src/components/App.js

```diff
@@ -1,15 +1,161 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { BrowserRouter, Route } from 'react-router-dom';
+import { getWeatherInformation, toCelcius, toFahrenheit } from '../utils/weather';
+import getLocation from '../utils/location';
+
+import UVIndex from './UVIndex';
+import Forecast from './Forecast';
+import Nav from './Nav';
+import UnitToggle from './UnitToggle';
 import Weather from './Weather';
 
-const App = () => (
-  <div>
-    <h1>React Boilerplate with Sass</h1>
-    <img
-      src="https://upload.wikimedia.org/wikipedia/commons/a/a7/React-icon.svg"
-      alt="react-logo"
-    />
-    <Weather />
-  </div>
-);
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      location: false,
+      latitude: null,
+      longitude: null,
+      weatherData: {},
+      forecastData: {},
+      forecastResponse: false,
+      weatherResponse: false,
+      uvResponse: false,
+      uvData: null,
+      temperature: {},
+    };
+
+    this.changeUnit = this.changeUnit.bind(this);
+  }
+
+  componentDidMount() {
+    if ('geolocation' in navigator) {
+      getLocation().then(response => {
+        this.setState({
+          location: true,
+          latitude: response.coords.latitude,
+          longitude: response.coords.longitude,
+        });
+        this.getCurrentWeather();
+        this.getCurrentForecast();
+        this.getUVIndex();
+      });
+    }
+  }
+
+  getCurrentWeather() {
+    const { latitude, longitude, location } = this.state;
+    if (location) {
+      getWeatherInformation('weather', latitude, longitude).then(response => {
+        const { data } = response;
+        this.setState({
+          weatherResponse: true,
+          temperature: { type: 'FAHRENHEIT', value: data.main.temp, unit: 'F' },
+          weatherData: {
+            name: data.name,
+            description: data.weather[0].description,
+            icon: data.weather[0].icon,
```

Comment:
** Next PR **
destructure your data in the api call function; then you'll isolate the api response details in one place. take a minute to look at what your app needs for data (probably roughly what this thing is doing)

Reply:
done